### ISSUE TITLE
Unable to access app deployed on Dokku in a droplet

### ISSUE THREAD

**Issue by hasannadeem** [dokku/dokku#4407]
## Description of problem
I'm deploying a Ruby on Rails 6.0.3 application using `dokku version 0.21.4`, and facing some issues. I've deployed my Dokku App on a Digital Ocean droplet using Dokku image, but I'm unable to access it. I've enabled dokku `virtualhost` for this droplet. I've followed [this guide for deployment](https://www.youtube.com/watch?v=E1P6k5_IKAs&t=4s), which works fine there, but I'm ending up with some blunder I guess. <br>
Have posted the issue on [stackoverflow](https://stackoverflow.com/questions/65958620/unable-to-access-deployed-app-from-digital-ocean-droplet?noredirect=1#comment116758264_65958620) and [digitalocean forums](https://www.digitalocean.com/community/questions/unable-to-access-app-deployed-on-dokku-in-a-droplet), but haven't been able to resolve this, as of now.

### How reproducible
I have tried this several times and ended with same result. It actually behaved same when I tried it without `virtualhost` setting enabled. 

### Steps to Reproduce

1. Create a Dokku app
2. Install dokku `postgres` plugin and link it to the dokku app (not related to the issue as far as I can tell)
3. Set up git repo on Dokku with `dokku git:initialize app-name`
4. Push the app for deployment to the Dokku instance 
5. Configure the domain name
6. Open `ufw` firewall for `http` and `https`
7. Try to hit the app from browser(outside dokku)

#### Actual Results
The app is not accessible outside of the Dokku. Whenever I visit the app by configured domain name, it shows
```
This site can’t be reached. safepropertymyanmar.com refused to connect.
Try:
Checking the connection
Checking the proxy and the firewall
ERR_CONNECTION_REFUSED
```

#### Expected Results
The deployed app should be accessible on the configured domain and respond to `http` and `https` requests.

### `dokku report app-name` output
Here is the output of `dokku report my-app-name`
```
-----> uname: Linux dokku-ubuntu-s-1vcpu-2gb-sgp1-01 5.4.0-52-generic #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
                     total        used        free      shared  buff/cache   available
       Mem:           1987         405         190          15        1391        1411
       Swap:             0           0           0
-----> docker version: 
       Client: Docker Engine - Community
        Version:           19.03.13
        API version:       1.40
        Go version:        go1.13.15
        Git commit:        4484c46d9d
        Built:             Wed Sep 16 17:02:52 2020
        OS/Arch:           linux/amd64
        Experimental:      false
       
       Server: Docker Engine - Community
        Engine:
         Version:          19.03.13
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.13.15
         Git commit:       4484c46d9d
         Built:            Wed Sep 16 17:01:20 2020
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.3.7
         GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
        runc:
         Version:          1.0.0-rc10
         GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
        docker-init:
         Version:          0.18.0
         GitCommit:        fec3683
-----> docker daemon info: 
       Client:
        Debug Mode: true
       
       Server:
        Containers: 9
         Running: 2
         Paused: 0
         Stopped: 7
        Images: 13
        Server Version: 19.03.13
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 8fba4e9a7d01810a393d5d25a3621dc101981175
        runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
        init version: fec3683
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 5.4.0-52-generic
        Operating System: Ubuntu 20.04.1 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 1
        Total Memory: 1.941GiB
        Name: dokku-ubuntu-s-1vcpu-2gb-sgp1-01
        ID: PEEB:CHZO:EMP7:XJR5:2ZB7:QM5F:TPDW:PHRP:MO73:X7YR:V5PN:LJDA
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false
       
-----> sigil version: 0.6.0
-----> herokuish version: 
       herokuish: 0.5.18
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v218
         heroku-buildpack-nodejs    v175
         heroku-buildpack-clojure   v85
         heroku-buildpack-python    v177
         heroku-buildpack-java      v66
         heroku-buildpack-gradle    v31
         heroku-buildpack-scala     v87
         heroku-buildpack-play      v26
         heroku-buildpack-php       v179
         heroku-buildpack-go        v145
         buildpack-nginx            v14
-----> dokku version: dokku version 0.21.4
-----> dokku plugins: 
       plugn: 0.5.0
         00_dokku-standard    0.21.4 enabled    dokku core standard plugin
         20_events            0.21.4 enabled    dokku core events logging plugin
         app-json             0.21.4 enabled    dokku core app-json plugin
         apps                 0.21.4 enabled    dokku core apps plugin
         builder-dockerfile   0.21.4 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.21.4 enabled    dokku core builder-herokuish plugin
         buildpacks           0.21.4 enabled    dokku core buildpacks plugin
         certs                0.21.4 enabled    dokku core certificate management plugin
         checks               0.21.4 enabled    dokku core checks plugin
         common               0.21.4 enabled    dokku core common plugin
         config               0.21.4 enabled    dokku core config plugin
         docker-options       0.21.4 enabled    dokku core docker-options plugin
         domains              0.21.4 enabled    dokku core domains plugin
         enter                0.21.4 enabled    dokku core enter plugin
         git                  0.21.4 enabled    dokku core git plugin
         logs                 0.21.4 enabled    dokku core logs plugin
         network              0.21.4 enabled    dokku core network plugin
         nginx-vhosts         0.21.4 enabled    dokku core nginx-vhosts plugin
         plugin               0.21.4 enabled    dokku core plugin plugin
         postgres             1.11.6 enabled    dokku postgres service plugin
         proxy                0.21.4 enabled    dokku core proxy plugin
         ps                   0.21.4 enabled    dokku core ps plugin
         repo                 0.21.4 enabled    dokku core repo plugin
         resource             0.21.4 enabled    dokku core resource plugin
         scheduler-docker-local 0.21.4 enabled    dokku core scheduler-docker-local plugin
         shell                0.21.4 enabled    dokku core shell plugin
         ssh-keys             0.21.4 enabled    dokku core ssh-keys plugin
         storage              0.21.4 enabled    dokku core storage plugin
         tags                 0.21.4 enabled    dokku core tags plugin
         tar                  0.21.4 enabled    dokku core tar plugin
         trace                0.21.4 enabled    dokku core trace plugin
=====> propertysafe app information
       App deploy source:             
       App dir:                       /home/dokku/propertysafe
       App locked:                    false
=====> propertysafe buildpacks information
       Buildpacks list:               
=====> propertysafe ssl information
       Ssl dir:                       /home/dokku/propertysafe/tls
       Ssl enabled:                   false                    
       Ssl hostnames:                                          
       Ssl expires at:                                         
       Ssl issuer:                                             
       Ssl starts at:                                          
       Ssl subject:                                            
       Ssl verified:                                           
=====> propertysafe checks information
       Checks disabled list:          none                     
       Checks skipped list:           none                     
=====> propertysafe docker options information
       Docker options build:          --link dokku.postgres.propertysafe_db:dokku-postgres-propertysafe-production 
       Docker options deploy:         --link dokku.postgres.propertysafe_db:dokku-postgres-propertysafe-production --restart=on-failure:10 
       Docker options run:            --link dokku.postgres.propertysafe_db:dokku-postgres-propertysafe-production 
=====> propertysafe domains information
       Domains app enabled:           true                     
       Domains app vhosts:            safepropertymyanmar.com  
       Domains global enabled:        false                    
       Domains global vhosts:                                  
=====> propertysafe git information
       Git deploy branch:             master                   
       Git global deploy branch:      master                   
       Git keep git dir:              false                    
       Git rev env var:               GIT_REV                  
       Git sha:                       6ab0e28                  
       Git last updated at:                                    
=====> propertysafe network information
       Network attach post create:    
       Network attach post deploy:    
       Network bind all interfaces:   false
       Network web listeners:         172.17.0.2:5000
=====> propertysafe nginx information
       Nginx access log path:         /var/log/nginx/propertysafe-access.log
       Nginx bind address ipv4:                                
       Nginx bind address ipv6:       ::                       
       Nginx error log path:          /var/log/nginx/propertysafe-error.log
       Nginx hsts:                    true                     
       Nginx hsts include subdomains: true                     
       Nginx hsts max age:            15724800                 
       Nginx hsts preload:            false                    
       Nginx proxy read timeout:      60s                      
       Nginx last visited at:         1613467086               
=====> propertysafe proxy information
       Proxy enabled:                 true
       Proxy port map:                http:80:5000
       Proxy type:                    nginx
=====> propertysafe ps information
       Processes:                     1                        
       Deployed:                      true                     
       Running:                       true                     
       Restore:                       true                     
       Restart policy:                on-failure:10            
       Ps can scale:                  true                     
       Status web.1:                  running    (CID: e6a46b43ca00)
=====> propertysafe scheduler-docker-local information
       Scheduler docker local disable chown:                          
=====> propertysafe storage information
       Storage build mounts:                                   
       Storage deploy mounts:                                  
       Storage run mounts:
```

### Additional information

- Container Inspect Output via `dokku ps:inspect app-name`

```
[
    {
        "AppArmorProfile": "docker-default",
        "Args": [
            "web"
        ],
        "Config": {
            "AttachStderr": true,
            "AttachStdin": false,
            "AttachStdout": true,
            "Cmd": [
                "/start",
                "web"
            ],
            "Domainname": "",
            "Entrypoint": null,
            "Env": [
                "USER=herokuishuser",
                "DYNO=web.1",
                "PORT=5000",
                "CACHE_PATH=/cache",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "DEBIAN_FRONTEND=XXXXXX"
            ],
            "Hostname": "e6a46b43ca00",
            "Image": "dokku/propertysafe:latest",
            "Labels": {
                "com.dokku.app-name": "propertysafe",
                "com.dokku.container-type": "deploy",
                "com.dokku.dyno": "web.1",
                "com.dokku.image-stage": "release",
                "com.dokku.process-type": "web",
                "dokku": "",
                "org.label-schema.schema-version": "1.0",
                "org.label-schema.vendor": "dokku"
            },
            "OnBuild": null,
            "OpenStdin": false,
            "StdinOnce": false,
            "Tty": false,
            "User": "",
            "Volumes": null,
            "WorkingDir": ""
        },
        "Created": "2021-02-16T09:07:15.72113511Z",
        "Driver": "overlay2",
        "ExecIDs": null,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/e3550558e88895af1149e52b9293bd37587dc91e3f7245d48c3e01d3c115661d-init/diff:/var/lib/docker/overlay2/07660c0a5fbca6083ddd0ac68fd39e0f0772993b991e87cc1d4f303bcba0f9db/diff:/var/lib/docker/overlay2/d0601c8428aec90420f40de74a561267636cb0b930723ec123ceb816337f1ba5/diff:/var/lib/docker/overlay2/ec2261bc13ab9b9ebb088aede5814474c91a68f1773f182a29d65fd18217d260/diff:/var/lib/docker/overlay2/6f64b2f987fe59f71a0301377352ff206093fe94f04f1dd3c7604c32639b5e06/diff:/var/lib/docker/overlay2/935a7f22af49431610fa8f78b80d74824db50d582f6f3f8203e3dd9ee37f5f49/diff:/var/lib/docker/overlay2/ea392844a1ce8f70fb3f2f04f814efe310646f38da7ddf1c94d80aa0004db968/diff:/var/lib/docker/overlay2/31b3536a78bdadc46cb88a3c50e0f6cf94fb7119eae9d2f31fc0e25c443872a7/diff:/var/lib/docker/overlay2/4efe79f3c127183fd96a6f188e68a40ac54d01655695ca6e7690b86847054bbc/diff:/var/lib/docker/overlay2/72811bb9c28dc15f741cfb8488877851d460cdbec376ad0f81eac6bfaf24363a/diff:/var/lib/docker/overlay2/d630dca098d5f22ffd6da90801c3054aa79805048c6b2dd47cd5acf48dc289a9/diff:/var/lib/docker/overlay2/ca4740f9464caede5d2a59688230f9709af2a9e63e5b81cdb2bc661230e93767/diff:/var/lib/docker/overlay2/40a71248114edfe0a803fa1cda3ff4da906acfab82d6b32901004fcb068360e4/diff:/var/lib/docker/overlay2/1d0613ea75384886e3b6b884a2eb61746b31d63d1ec040f6bdcced489563c0f0/diff:/var/lib/docker/overlay2/bf0fdaf11d50bdb1381a9b932e6d1fa7456372d02d4a1ece020342f46ce53b82/diff:/var/lib/docker/overlay2/f4ef365a7df50734d1a078a220b8f8731eab0e5f3144bb85cbb8f27f31728fa3/diff:/var/lib/docker/overlay2/1446232250fcb3100c41169ed84eec17b9974c4d6560152d60c928302afc08c9/diff:/var/lib/docker/overlay2/4784066a41fe0a68828dae2130b3a6837beea10d6dacce5cf67234f5502591a7/diff:/var/lib/docker/overlay2/1bf868816b5528c4ab961cecda728b9c34fc3c2bcddef90f43e3ad2bd56f80ae/diff:/var/lib/docker/overlay2/ca47f04e993f4676bde48e6cd3fe8062e1111b4cbcfc6f9c5479f4114d9c4ca4/diff",
                "MergedDir": "/var/lib/docker/overlay2/e3550558e88895af1149e52b9293bd37587dc91e3f7245d48c3e01d3c115661d/merged",
                "UpperDir": "/var/lib/docker/overlay2/e3550558e88895af1149e52b9293bd37587dc91e3f7245d48c3e01d3c115661d/diff",
                "WorkDir": "/var/lib/docker/overlay2/e3550558e88895af1149e52b9293bd37587dc91e3f7245d48c3e01d3c115661d/work"
            },
            "Name": "overlay2"
        },
        "HostConfig": {
            "AutoRemove": false,
            "Binds": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceWriteIOps": null,
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "CapAdd": null,
            "CapDrop": null,
            "Capabilities": null,
            "Cgroup": "",
            "CgroupParent": "",
            "ConsoleSize": [
                0,
                0
            ],
            "ContainerIDFile": "",
            "CpuCount": 0,
            "CpuPercent": 0,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpuShares": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "Devices": [],
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IOMaximumBandwidth": 0,
            "IOMaximumIOps": 0,
            "Init": true,
            "IpcMode": "private",
            "Isolation": "",
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "Links": [
                "/dokku.postgres.propertysafe_db:/propertysafe.web.1/dokku-postgres-propertysafe-production"
            ],
            "LogConfig": {
                "Config": {},
                "Type": "json-file"
            },
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "Memory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "NanoCpus": 0,
            "NetworkMode": "default",
            "OomKillDisable": false,
            "OomScoreAdj": 0,
            "PidMode": "",
            "PidsLimit": null,
            "PortBindings": {},
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ],
            "ReadonlyRootfs": false,
            "RestartPolicy": {
                "MaximumRetryCount": 10,
                "Name": "on-failure"
            },
            "Runtime": "runc",
            "SecurityOpt": null,
            "ShmSize": 67108864,
            "UTSMode": "",
            "Ulimits": null,
            "UsernsMode": "",
            "VolumeDriver": "",
            "VolumesFrom": null
        },
        "HostnamePath": "/var/lib/docker/containers/e6a46b43ca00d012fa0e49effa08d4d6e02fef10e6b8d3753958cca3dd2bbed1/hostname",
        "HostsPath": "/var/lib/docker/containers/e6a46b43ca00d012fa0e49effa08d4d6e02fef10e6b8d3753958cca3dd2bbed1/hosts",
        "Id": "e6a46b43ca00d012fa0e49effa08d4d6e02fef10e6b8d3753958cca3dd2bbed1",
        "Image": "sha256:aaa84d426a714dec929c68e59a0765f5be590890dd25e3e988af205fa2e99027",
        "LogPath": "/var/lib/docker/containers/e6a46b43ca00d012fa0e49effa08d4d6e02fef10e6b8d3753958cca3dd2bbed1/e6a46b43ca00d012fa0e49effa08d4d6e02fef10e6b8d3753958cca3dd2bbed1-json.log",
        "MountLabel": "",
        "Mounts": [],
        "Name": "/propertysafe.web.1",
        "NetworkSettings": {
            "Bridge": "",
            "EndpointID": "55d8558ac5348d3a7e9a416b276b5d3f5e195b263c5aba8818eb725428d3bf74",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "HairpinMode": false,
            "IPAddress": "172.17.0.2",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "MacAddress": "02:42:ac:11:00:02",
            "Networks": {
                "bridge": {
                    "Aliases": null,
                    "DriverOpts": null,
                    "EndpointID": "55d8558ac5348d3a7e9a416b276b5d3f5e195b263c5aba8818eb725428d3bf74",
                    "Gateway": "172.17.0.1",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "IPAMConfig": null,
                    "IPAddress": "172.17.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "Links": null,
                    "MacAddress": "02:42:ac:11:00:02",
                    "NetworkID": "22d4a3c222ee7cc3c9c6b8318a19333498655914775e49a90fcfa1f16edc99c6"
                }
            },
            "Ports": {},
            "SandboxID": "b11103389c357867b3717f71a80ee073b0938e91251e876832897b7090605ea3",
            "SandboxKey": "/var/run/docker/netns/b11103389c35",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null
        },
        "Path": "/start",
        "Platform": "linux",
        "ProcessLabel": "",
        "ResolvConfPath": "/var/lib/docker/containers/e6a46b43ca00d012fa0e49effa08d4d6e02fef10e6b8d3753958cca3dd2bbed1/resolv.conf",
        "RestartCount": 0,
        "State": {
            "Dead": false,
            "Error": "",
            "ExitCode": 0,
            "FinishedAt": "0001-01-01T00:00:00Z",
            "OOMKilled": false,
            "Paused": false,
            "Pid": 123688,
            "Restarting": false,
            "Running": true,
            "StartedAt": "2021-02-16T09:07:16.504596102Z",
            "Status": "running"
        }
    }
]
```
- The nginx configuration (if applicable) via `dokku nginx:show-config my-app-name`
```
server {
  listen      [::]:80;
  listen      80;
  server_name safepropertymyanmar.com; 
  access_log  /var/log/nginx/propertysafe-access.log;
  error_log   /var/log/nginx/propertysafe-error.log;

  location    / {

    gzip on;
    gzip_min_length  1100;
    gzip_buffers  4 32k;
    gzip_types    text/css text/javascript text/xml text/plain text/x-component application/javascript application/x-javascript application/json application/xml  application/rss+xml font/truetype application/x-font-ttf font/opentype application/vnd.ms-fontobject image/svg+xml;
    gzip_vary on;
    gzip_comp_level  6;

    proxy_pass  http://propertysafe-5000;
    proxy_http_version 1.1;
    proxy_read_timeout 60s;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_set_header Host $http_host;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-For $remote_addr;
    proxy_set_header X-Forwarded-Port $server_port;
    proxy_set_header X-Request-Start $msec;
  }
  include /home/dokku/propertysafe/nginx.conf.d/*.conf;

  error_page 400 401 402 403 405 406 407 408 409 410 411 412 413 414 415 416 417 418 420 422 423 424 426 428 429 431 444 449 450 451 /400-error.html;
  location /400-error.html {
    root /var/lib/dokku/data/nginx-vhosts/dokku-errors;
    internal;
  }

  error_page 404 /404-error.html;
  location /404-error.html {
    root /var/lib/dokku/data/nginx-vhosts/dokku-errors;
    internal;
  }

  error_page 500 501 502 503 504 505 506 507 508 509 510 511 /500-error.html;
  location /500-error.html {
    root /var/lib/dokku/data/nginx-vhosts/dokku-errors;
    internal;
  }

}

upstream propertysafe-5000 {

  server 172.17.0.2:5000;
}

```
- Output of `docker ps`
```
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS              PORTS               NAMES
e6a46b43ca00        dokku/propertysafe:latest   "/start web"             28 minutes ago      Up 28 minutes                           propertysafe.web.1
aebebea59bfb        postgres:11.6               "docker-entrypoint.s…"   About an hour ago   Up About an hour    5432/tcp            dokku.postgres.propertysafe_db
```
I believe here is the real issue, the Docker Container which has my app has not exposed its port to the Dokku host.
I ran  `docker port e6a46b43ca00` and it threw this error
```Error: No public port '5000/tcp' published for e6a46b43ca00```
and hence the request might never be reaching the Docker Container's port 5000, where my `rails server` is running.

- Dokku vhost domain config
Output of `dokku domains:report app-name`
```
=====> propertysafe domains information
       Domains app enabled:           true                     
       Domains app vhosts:            safepropertymyanmar.com  
       Domains global enabled:        false                    
       Domains global vhosts:              
```
- Dokku port mappings
Output of `dokku proxy:report`
```
=====> propertysafe proxy information
       Proxy enabled:                 true
       Proxy port map:                http:80:5000
       Proxy type:                    nginx
```
- Output of `sudo netstat -plant`
```
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      2847/nginx: master  
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      620/systemd-resolve 
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      836/sshd: /usr/sbin 
tcp        0    400 128.199.191.76:22       72.255.43.70:14704      ESTABLISHED 3191/sshd: root@pts 
tcp        0      0 128.199.191.76:80       192.241.204.244:41173   SYN_RECV    -                   
tcp6       0      0 :::80                   :::*                    LISTEN      2847/nginx: master  
tcp6       0      0 :::22                   :::*                    LISTEN      836/sshd: /usr/sbin
```
- Firewall status
Output of `sudo ufw status`
```
Status: active

To                         Action      From
--                         ------      ----
22/tcp                     LIMIT       Anywhere                  
2375/tcp                   ALLOW       Anywhere                  
2376/tcp                   ALLOW       Anywhere                  
Nginx Full                 ALLOW       Anywhere                  
80/tcp                     ALLOW       Anywhere                  
443/tcp                    ALLOW       Anywhere                  
22/tcp (v6)                LIMIT       Anywhere (v6)             
2375/tcp (v6)              ALLOW       Anywhere (v6)             
2376/tcp (v6)              ALLOW       Anywhere (v6)             
Nginx Full (v6)            ALLOW       Anywhere (v6)             
80/tcp (v6)                ALLOW       Anywhere (v6)             
443/tcp (v6)               ALLOW       Anywhere (v6)
```

- Rails application's `config/puma.rb` file
It's content is as follows
```
max_threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }
min_threads_count = ENV.fetch("RAILS_MIN_THREADS") { max_threads_count }
threads min_threads_count, max_threads_count

# Specifies the `port` that Puma will listen on to receive requests; default is 3000.
#
port        ENV.fetch("PORT") { 3000 }

# Specifies the `environment` that Puma will run in.
#
environment ENV.fetch("RAILS_ENV") { "development" }

# Specifies the `pidfile` that Puma will use.
pidfile ENV.fetch("PIDFILE") { "tmp/pids/server.pid" }

plugin :tmp_restart
```
- Procfile
```
web: bundle exec puma -C config/puma.rb
```

**Comment by josegonzalez** [comment]
What is the IP address for your server? Also, can you show the output of `dokku logs $APP`, where `$APP` is your app's name?

**Comment by hasannadeem** [comment]
My droplet's IP address is 128.199.191.76
Output of `dokku logs app-name`
```
2021-02-16T09:07:18.482117382Z app[web.1]: Puma starting in single mode...
2021-02-16T09:07:18.482560281Z app[web.1]: * Version 4.3.5 (ruby 2.7.0-p0), codename: Mysterious Traveller
2021-02-16T09:07:18.482656371Z app[web.1]: * Min threads: 5, max threads: 5
2021-02-16T09:07:18.482742865Z app[web.1]: * Environment: production
2021-02-16T09:07:22.407761221Z app[web.1]: D, [2021-02-16T09:07:22.407462 #10] DEBUG -- : BONSAI_URL not present, proceeding with Elasticsearch defaults.
2021-02-16T09:07:24.609457780Z app[web.1]: * Listening on tcp://0.0.0.0:5000
2021-02-16T09:07:24.610035928Z app[web.1]: Use Ctrl-C to stop
2021-02-16T23:13:29.778558169Z app[web.1]: 2021-02-16 23:13:29 +0000: HTTP parse error, malformed request (/): #<Puma::HttpParserError: Invalid HTTP format, parsing fails.>
2021-02-16T23:13:29.778587733Z app[web.1]: ---
```
Http/https requests are not reaching to the rails server running at port 5000 of the container.

**Comment by josegonzalez** [comment]
When I try curling that IP with your domain, I get the following:

```
% curl --include -H "Host: safepropertymyanmar.com" 128.199.191.76
HTTP/1.1 301 Moved Permanently
Server: nginx
Date: Wed, 17 Feb 2021 06:01:17 GMT
Content-Type: text/html
Transfer-Encoding: chunked
Connection: keep-alive
Location: https://safepropertymyanmar.com/
```

Curling the 443 version fails (i've used `-k` below to override any ssl warnings:

```
% curl --include -H "Host: safepropertymyanmar.com" -k https://128.199.191.76
curl: (7) Failed to connect to 128.199.191.76 port 443: Connection refused
```

Two thoughts:
- Do you have cloudflare or something else in front of the app? If so, can you disable any ssl forwarding enabled on that?
- Does the rails app potentially force enable ssl? If so, can you disable that in the code and redeploy?

> Aside: the dokku app is now git initialized on creation, so running `dokku git:initialize app-name` is no longer necessary.

**Comment by hasannadeem** [comment]
Yes, It finally gets resolved. You're right abut the forced ssl part. My rails app had enabled force_ssl config.
- I did the following in `config/environments/production.rb`
```
config.force_ssl = false
```
and deployed it again.

- This step might not have been need but I did it anyways. On Dokku console, ran the following command to reload nginx service
`sudo systemctl reload nginx`

- Visited the URL/IP using `http` protocol and it worked like a charm.

Thank you very much for your time and effort. This issue can be closed now.
